src/htcp.cc

```diff
@@ -762,7 +803,13 @@ htcpUnpackDetail(char *buf, int sz)
     sz -= l;
 
     /* Find length of CACHE-HDRS */
-    l = ntohs(*(uint16_t *) buf);
+    if (sz < 2) {
+        debugs(31, 3, "htcpUnpackDetail: too short for CACHE-HDRS length");
+        delete d;
+        return nullptr;
+    }
+    memcpy(&l, buf, 2);
+    l = ntohs(l);
```

Comment:
Please avoid creating so many duplicates of this (now correct) "parse short integer" code. Add a `parseUint16(buf, size, description)` or similar static function. Use std::unique_ptr to avoid deleting `d` manually on errors.